### PR TITLE
Task#29 - Feat: Display Total Cost

### LINKED ISSUE #100: User Story#3: Adjust Quantity and Display Total Cost
> As a user, I want to adjust the number of items in my cart and see the total cost before proceeding to payment.

- [x] #101

- [x] When adding products already in the box, the quality of the products will also be increased.
- [x] Within the cart summary sidebar, provide controls to increase or decrease the quantity of each product.
- [x] Ensure that adjusting quantities also update the corresponding total cost for each product.
- [x] When the quantity of the product's quantity reaches 0, the product will be removed from the cart

- [x] #104

- [x] Display the calculated total cost of all products in the cart at the bottom of the sidebar.
- [x] Update the total cost dynamically as users adjust quantities.

Task#3: Test Quantity Adjustment and Total Cost Calculation

- [x] Test the quantity adjustment controls for accuracy and responsiveness.
- [x] Verify that the total cost accurately reflects the adjusted quantities.

### LINKED ISSUE #104: Task#29 - Dev Task: Display Total Cost
- #89 
  - #100 
## OVERVIEW
- Display the price of the product according to the quantity
- Update the total price of the product according to the price of the product in the cart
## DEMO

https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/61428348-79c2-436c-bdeb-780bc4e53a75

### PR DESCRIPTION
**Pull request by TheHienDuong**
- Closes #104 

## OVERVIEW
- Create updateProductPriceAndQuantity function
  - Update price when quantity changes
- Create a filter that converts an amount to currency, and calculates Total Prices
- Get ProductPrice to calculate the total value of the products in the cart
- Update the totalPrice component to handle the total price

## CHECKLIST
- [x] Code style: Drop the line before each comment and block the code

## DEMO
https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/79547a42-9488-4de5-b689-2976749c7082

### CODE REVIEW

File: practice-two/src/components/CartItem.jsx [hunk 1]
```diff
@@ -53,35 +56,65 @@ const ProductImage = ({ src, alt }) => (
...
+  // Function to increase the product quantity by 1
   const increaseQuantity = () => {
-    const updatedQuantity = quantity + 1;
-    setQuantity(updatedQuantity);
-    updateQuantity(product.id, updatedQuantity);
+    updateProductPriceAndQuantity(quantity + 1);
   };
 
+  // UseEffect hook to update the product's total price whenever the quantity changes
+  useEffect(() => {
```
Comment (phattruong-agilityio) [inline practice-two/src/components/CartItem.jsx]: I don't find any reason why we need to use the useEffect, Please help check , ping me if you need help

File: practice-two/src/components/ShoppingCart.jsx [hunk 2]
```diff
@@ -29,8 +29,12 @@ const ShoppingCart = ({
   removeFromCart = () => {},
   updateQuantity = () => {},
 }) => {
+  // State to manage the status of the slider bar (open/closed)
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ShoppingCart.jsx:35]: code style

File: practice-two/src/components/ShoppingCart.jsx [hunk 3]
```diff
@@ -41,6 +45,15 @@ const ShoppingCart = ({
     document.body.classList.remove('overflow-hidden');
   };
 
+  // Update the stored price for a specific product
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ShoppingCart.jsx]: Is this format correct

File: practice-two/src/components/TotalSection.jsx [hunk 4]
```diff
@@ -1,38 +1,50 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
 // Components
 import Typography from './common/Typography';
 
+// Filter
```
Comment (phattruong-agilityio) [inline practice-two/src/components/TotalSection.jsx]: useless comment

**Review by phattruong-agilityio — APPROVED** [review]

File: practice-two/src/components/TotalSection.jsx (hunk 4 above)
Comment (TheHienDuong) [reply practice-two/src/components/TotalSection.jsx]: Update into // Helper functions

File: practice-two/src/components/ShoppingCart.jsx (hunk 2 above)
Comment (TheHienDuong) [reply practice-two/src/components/ShoppingCart.jsx:35]: I'm having some code style issues here, formatting by prettier and eslint both disallow spaces before comments.
Specifically, if you turn off the rule of eslint, the formatting will also automatically align the space.

## DEMO

https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/b6a3ceec-f2ba-4606-b13f-bcee496f8a35

File: practice-two/src/components/CartItem.jsx (hunk 1 above)
Comment (TheHienDuong) [reply practice-two/src/components/CartItem.jsx]: useEffect helps to update prices when products are added to the cart or quantity changes.
When usingEffect is not used when adding a product to the cart, the total cost will not be updated, only updated when the user increases or decreases the quantity of the product.

## DEMO

https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/ef012d70-60f1-4fc4-8f3e-78b2f0c704d2

**Comment by TheHienDuong** [comment]
- This is an error, add to cart will be added and paid for the product the first time, Increase the quantity in the cart without any problem, until the same product click added to the cart again. The value of that product will be price * quantity = 1 (even if the quantity is 5 )
- Tried to pass prop as quantity when updated and added (quantity, when added and increase and decrease, are both normal) and to getProductPrice ( price * quantity ), but can't fix it

## DEMO
https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/5ff5845c-d5e5-4b43-81e1-2c392ae65b25

**Review by phattruong-agilityio — APPROVED** [review]

**Pull request merged**